sec_certs/dataset.py

```diff
@@ -1210,18 +1258,33 @@ def split_alg(alg_string):
                 html_soup = BeautifulSoup(handle.read(), 'html.parser')
 
             table = html_soup.find('table', class_='table table-condensed publications-table table-bordered')
-            spans = table.find_all('span')
-            for span in spans:
-                elements = span.find_all('td')
-                vendor, implementation = elements[0].text, elements[1].text
-                elements_sliced = elements[2:]
-                for i in range(0, len(elements_sliced), 2):
-                    alg_type, alg_id = split_alg(elements_sliced[i].text.strip())
-                    validation_date = elements_sliced[i + 1].text.strip()
-                    fips_alg = FIPSCertificate.Algorithm(alg_id, vendor, implementation, alg_type, validation_date)
-                    if alg_id not in self.certs:
-                        self.certs[alg_id] = []
-                    self.certs[alg_id].append(fips_alg)
+            tbody_contents = table.find('tbody').find_all('tr')
+            current_vendor = current_product = current_validation = current_date = ""
+            for tr in tbody_contents:
```

Comment:
The cyclomatic complexity of this function is really high, I suggest refactoring it. Also, consider using conditional expressions like
```python
current_validation = elem.text.strip() if 'validation-number' in attachment['id'] else ''
```